keps/sig-storage/1698-generic-ephemeral-volumes/README.md

```diff
@@ -0,0 +1,566 @@
+light-weight, local volumes. It is not a good fit for volumes provided
+by a more traditional storage system because:
+- The normal API for selecting volume parameters (like size and
+  storage class) is not supported.
+- Integration into storage capacity aware pod scheduling is
+  challenging and would depend on extending the CSI ephemeral inline
+  volume API.
+- CSI drivers need to be adapted and have to take over some of the
+  work normally done by Kubernetes, like tracking of orphaned volumes.
+
+
+### Goals
+
+- Volumes can be specified inside the pod spec ("inline").
+- Volumes are created for specific pods and deleted after the pod
+  terminates ("ephemeral").
+- A normal, unmodified storage driver can be selected via a storage class.
+- The volume will be created using the normal storage provisioning
+  mechanism, without having to modify the driver or its deployment.
+- Storage capacity tracking can be enabled also for such volumes.
+
+### Non-Goals
+
+- This will not replace CSI ephemeral inline volumes because the
+  goals for those (light-weight local volumes) are not a good fit
+  for the approach proposed here.
+- These inline volumes will always be ephemeral. If making them persistent
+  was allowed, some additional controller would be needed to manage them
+  after pod termination, in which case it will probably be simpler to
+  also create them separately.
+
+## Proposal
+
+A new volume source will be introduced:
+
+```
+type EphemeralVolumeSource struct {
+    // Will be created as a stand-alone PVC to provision the volume.
+    // Required, must not be nil.
+    VolumeClaimTemplate *PersistentVolumeClaim
+    ReadOnly             bool
+}
+```
+
+This mimics a
+[`PersistentVolumeClaimVolumeSource`](https://pkg.go.dev/k8s.io/api/core/v1?tab=doc#PersistentVolumeClaimVolumeSource),
+except that it contains a `PersistentVolumeClaim` instead of
+referencing one by name.  A full `PersistentVolumeClaim` is used for
+consistency with the StatefulSet API.
+
+`VolumeClaimTemplate` is defined as a pointer because in the future there
+might be alternative ways to specify how the ephemeral
+volume gets provisioned, in which case `nil` will become a valid value.
+
+It gets embedded in the existing [`VolumeSource`
+struct](https://pkg.go.dev/k8s.io/api/core/v1?tab=doc#VolumeSource)
+struct as another alternative to the other ways of providing the
+volume, like
+[`CSIVolumeSource`](https://pkg.go.dev/k8s.io/api/core/v1?tab=doc#CSIVolumeSource)
+and
+[`PersistentVolumeClaimVolumeSource`](https://pkg.go.dev/k8s.io/api/core/v1?tab=doc#PersistentVolumeClaimVolumeSource).
+
+```
+type VolumeSource struct {
+   ...
+   CSI                   *CSIVolumeSource
+   PersistentVolumeClaim *PersistentVolumeClaimVolumeSource
+   ...
+   Ephemeral             *EphemeralVolumeSource
+   ...
+}
+```
+
+A new controller in `kube-controller-manager` is responsible for
+creating new PVCs for each such inline volume. It does that:
```

Comment:
ephemeral

Reply:
Fixed.

Let's hope this is the last one :smile:  :crossed_fingers: